### PR TITLE
Refs #9162 - removed compatibility workarounds

### PR DESCRIPTION
**Pull request by lzap**
This goes into **2.0-stable**.

### CODE REVIEW

File: lib/foreman_discovery/engine.rb [hunk 1]
```diff
@@ -119,13 +113,6 @@ class Engine < ::Rails::Engine
     end
 
     initializer "foreman_discovery.apipie" do
-      # this condition is here for compatibility reason to work with Foreman 1.4.x
-      # Also need to handle the reverse of the 1.8 method above
-      unless (SETTINGS[:version].to_s.include?('develop') or Gem::Version.new(SETTINGS[:version]) >= Gem::Version.new('1.8'))
-        if Apipie.configuration.api_controllers_matcher.is_a?(Array)
-          Apipie.configuration.api_controllers_matcher << "#{ForemanDiscovery::Engine.root}/app/controllers/api/v2/*.rb"
```
Comment (domcleal) [inline lib/foreman_discovery/engine.rb:126]: Don't you need this on 1.7?

**Comment by lzap** [comment]
@GregSutcliffe there are some concerns above, you have tested this you said on IRC. Can you comment on this?

**Comment by lzap** [comment]
Will open another one.

File: lib/foreman_discovery/engine.rb (hunk 1 above)
Comment (lzap) [inline lib/foreman_discovery/engine.rb:126]: Yeah I miss-read the `unless` block. Thought it was `if`. There is a new PR opened.

**Pull request closed without merging**